lib/util/extractSourceMap/parseDataURL.js

```diff
@@ -0,0 +1,463 @@
+		/** @type {string | undefined} */
+		subtype: undefined,
+		parameters: new Map(),
+		isBase64: Boolean(mimeTypeBase64MatchResult),
+		body
+	};
+
+	if (!mediaType) {
+		return result;
+	}
+
+	const inputMediaType = removeLeadingAndTrailingHTTPWhitespace(mediaType);
+
+	let positionMediaType = 0;
+	let type = "";
+
+	while (
+		positionMediaType < inputMediaType.length &&
+		inputMediaType[positionMediaType] !== "/"
+	) {
+		type += inputMediaType[positionMediaType];
+		positionMediaType += 1;
+	}
+
+	if (type.length === 0 || !solelyContainsHTTPTokenCodePoints(type)) {
+		return result;
+	}
+
+	if (positionMediaType >= inputMediaType.length) {
+		return result;
+	}
+
+	// Skips past "/"
+	positionMediaType += 1;
+
+	let subtype = "";
+
+	while (
+		positionMediaType < inputMediaType.length &&
+		inputMediaType[positionMediaType] !== ";"
+	) {
+		subtype += inputMediaType[positionMediaType];
+		positionMediaType += 1;
+	}
+
+	subtype = removeTrailingHTTPWhitespace(subtype);
+
+	if (subtype.length === 0 || !solelyContainsHTTPTokenCodePoints(subtype)) {
+		return result;
+	}
+
+	result.type = /** @type {string} */ (asciiLowercase(type));
+	result.subtype = /** @type {string} */ (asciiLowercase(subtype));
+
+	while (positionMediaType < inputMediaType.length) {
+		// Skip past ";"
+		positionMediaType += 1;
+
+		while (isHTTPWhitespaceChar(inputMediaType[positionMediaType])) {
+			positionMediaType += 1;
+		}
+
+		let parameterName = "";
+
+		while (
+			positionMediaType < inputMediaType.length &&
+			inputMediaType[positionMediaType] !== ";" &&
+			inputMediaType[positionMediaType] !== "="
+		) {
+			parameterName += inputMediaType[positionMediaType];
+			positionMediaType += 1;
+		}
+
+		parameterName = asciiLowercase(parameterName);
+
+		if (positionMediaType < inputMediaType.length) {
+			if (inputMediaType[positionMediaType] === ";") {
+				continue;
+			}
+
+			// Skip past "="
+			positionMediaType += 1;
+		}
+
+		let parameterValue = "";
+
+		if (inputMediaType[positionMediaType] === '"') {
+			[parameterValue, positionMediaType] = /** @type {[string, number]} */ (
+				collectAnHTTPQuotedString(inputMediaType, positionMediaType)
+			);
+
+			while (
+				positionMediaType < inputMediaType.length &&
+				inputMediaType[positionMediaType] !== ";"
+			) {
+				positionMediaType += 1;
+			}
+		} else {
+			while (
+				positionMediaType < inputMediaType.length &&
+				inputMediaType[positionMediaType] !== ";"
+			) {
+				parameterValue += inputMediaType[positionMediaType];
+				positionMediaType += 1;
+			}
+
+			parameterValue = removeTrailingHTTPWhitespace(parameterValue);
+
+			if (parameterValue === "") {
+				continue;
+			}
+		}
+
+		if (
+			parameterName.length > 0 &&
+			solelyContainsHTTPTokenCodePoints(parameterName) &&
+			solelyContainsHTTPQuotedStringTokenCodePoints(parameterValue) &&
+			!result.parameters.has(parameterName)
+		) {
+			result.parameters.set(parameterName, parameterValue);
+		}
+	}
+
+	return result;
+};
```

Comment:
Also currently we have the logic to parse `data:` URLs in `lib/schemes/DataUriPlugin.js` (we use regexp there), so I think we can union these logic and `DataUriPlugin.js` logic.

Before it let's benchmark it manually (maybe in future we will automatize such things) and how many tests were failed. And look who is win. 

We have a lot of tests - https://github.com/webpack-contrib/source-map-loader/blob/master/test/parseDataUrl.test.js for benchmarks

So:
- let's add a new `utils` - `parseDataURL`, 
- add unit tests from - https://github.com/webpack-contrib/source-map-loader/blob/master/test/parseDataUrl.test.js
- reuse this util in `lib/schemes/DataUriPlugin.js` and here
- decide which logic is more valid and faster after benchmark

We will have less code and reuse existing